ca/ca.go

```diff
@@ -252,6 +267,19 @@ func NewCertificateAuthorityImpl(
 	return ca, nil
 }
 
+func ResetSignatureCountForTesting() {
+	signatureCount.Reset()
+}
+
+func SignatureCountForTesting(certType string) int {
+	ch := make(chan prometheus.Metric, 10)
+	signatureCount.With(prometheus.Labels{"purpose": certType}).Collect(ch)
+	m := <-ch
+	var iom io_prometheus_client.Metric
+	_ = m.Write(&iom)
+	return int(iom.Counter.GetValue())
+}
```

Comment:
Remember that Go works on exported vs unexported; all variables and types, including unexported ones, are visible within the same package. So we can move these functions into ca_test.go without needed to export anything.

This approach (referencing the global variable directly, rather than creating a field to be overridden) is indeed nicer. I didn't realize you can collect the output of a metric limited to certain label-value pairs. That is quite handy.